### ISSUE TITLE
Make sure that web resources handled by the Vaadin Servlet are available in OSGi

### ISSUE THREAD

**Issue by denis-anisimov** [vaadin/flow#4229]
This ticket is about resources which are internally handled by the Vaadin servlet as a static resource (via `StaticFileServer`).
There are some internal URLs which are handled by Flow like `StreamResource` (their URI starts with `VAADIN/dynamic/resource/`) see `StreamRequestHandler::generateURI` .
Also VaadinServlet has logic which handles webjar URLs.
Those resources should be available as web resources in OSGi.

See https://github.com/mmerruko/framework/commit/687a9009cf3bd2852181848c0c466aac5fe8f5d3 for related (but not necessary the same) support in FW8.

This will require some OSGi test environment. So it might be that this should not be done before https://github.com/vaadin/flow/issues/4227.

**Comment by denis-anisimov** [comment]
This is kind of blocked by the #4227 but it doesn't have to wait when the task will be completely done (and merged).
The #4227 should give a way to check whether an application can be used in OSGi.
At least there should be a simple way to deploy an application and check the result.
Otherwise this needs to be done every time when any OSGi related task is done and this is waste of time to reinvent the testing path every time.

**Comment by denis-anisimov** [comment]
What needs to be done: create an application, deploy it to OSGi and make sure the static web resources are available there.
If they custom way to do this is too complicated or may be improved then provide a way which can be used to simplify this in OSGi environment (see mentioned commit in FW).

**Comment by denis-anisimov** [comment]
Similar issue is #4230

**Comment by denis-anisimov** [comment]
The resolution comments for this ticket:
- webjars don't work at all with OSGi. Separate ticket for this: #4413.
- static resource has separate issue : #4230. 
  - Compiled client resources are registered via an `HttpService` (and so made available via HTTP).
  - Other static resources has an issue with `ServletContext`. It's describe in #4230.
- Dynamic internal resources functionality is tested via Templates test but there is no dedicated test for this. I will create one.

So in the result: this issue is divided to a number of sub-issues already reported.
The only remaining part here is dedicated test for dynamic resources.
